File: Tools/Board.js

function Board(rows, columns)
{
    this.rows = rows;
    this.columns = columns;

    this.boardC = [];

    this.Fruits = [];
}

Board.prototype.Columns = function() 
{ 
    return this.columns; 
}

Board.prototype.Rows = function()
{ 
    return this.rows;
}

Board.prototype.index = function( x, y)
{
    
    if(!Toolsc.In(x,y,this.rows,this.columns))
        throw  new Exception("Fuera");

    return this.boardC[x, y];
    
}

Board.prototype.setindex = function( x, y,value)
{
    
    if(!Toolsc.In(x,y,this.rows,this.columns))
        throw  new Exception("Fuera");

    this.boardC[x, y] = value;
    
}


Board.prototype.GenerateFruit = function(s)
{
    var xFruit = 0;
    var yFruit = 0;

    var t = new Toolsc();

    do
    {
        xFruit = t.getRandomInt(this.rows);
        yFruit = t.getRandomInt(this.columns);

    } while (this.Verify(xFruit,yFruit,s));

    this.Fruits.push(new Position(xFruit, yFruit));
}

Board.prototype.Fruits = function() 
{ 
    return this.Fruits;
}

Board.prototype.Verify = function( xFruit,  yFruit,  snake)
{
    for ( c in snake.Body)
    {
        var i = snake.Body[c];

        if (i.X == xFruit && i.Y == yFruit)
            return true;
    }

    for ( f in this.Fruits)
    {
        var fr = this.Fruits[f];

        if (fr.X == xFruit && fr.Y == yFruit)
            return true;
    }

    return false;
}

File: Snake/Snake/Logic/Board.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Snake.Logic
{
    public class Board
    {
        private int[,] boardC;

        public Board(int rows, int columns)
        {
            this.Rows = rows;
            this.Columns = columns;

            boardC = new int[rows,columns];

            Fruits = new List<Position>();
        }

        public int Columns { get; set; }

        public int Rows { get; set; }

        public object this[int x,int y]
        {
            get
            {
                if(!Tools.In(x,y,Rows,Columns))
                    throw  new Exception("Fuera");

                return boardC[x, y];
            }
            set
            {
                if (!Tools.In(x, y, Rows, Columns))
                    throw new Exception("Fuera");

                boardC[x, y] = (int)value;
            }
        }


        public void GenerateFruit(Snake s)
        {
            Random r = new Random(Environment.TickCount);

            int xFruit = 0;
            int yFruit = 0;

            do
            {
                xFruit = r.Next(0, Rows);
                yFruit = r.Next(0, Columns);

            } while (Verify(xFruit,yFruit,s));

            Fruits.Add(new Position(xFruit, yFruit));
        }

        public List<Position> Fruits { get; set; }

        private bool Verify(int xFruit, int yFruit, Snake snake)
        {
            foreach (var i in snake.Body)
            {
                if (i.X == xFruit && i.Y == yFruit)
                    return true;
            }

            return false;
        }
    }
}
